### DISCUSSION TITLE
Customise re-render with useTina() hook and/or "Save" button

### DISCUSSION THREAD

**Discussion by adrianfocke** [tinacms/tinacms#4436] [Help]
TL;DR: How can I change a state by clicking the save button from tina?

Dear tina community.

In my component, I use the [useTina](https://tina.io/docs/contextual-editing/react/#the-object-object-hook) hook with it's data object to patch a document. Normally, I would just add "data" to the dependency array of my useEffect hook to trigger some patch function whenever data changes. This time though, I do not want my function within the useEffect (the one that patches my document) to fire on every data change. What I did instead is to listen to a state which knows when to trigger the useEffect (as to be seen in the code below). This works perfectly fine, I am happy with that, but I need some more customisation.

This is what I need:

An event listener or maybe some tina-specific helper function (?) to know when the the blue tina iframe button that says "Save" is being clicked (or the graphql mutation is being sent). I can then automatically change the state to true, hence trigger my effect and maybe even force a re-render. Is there some solution for this, maybe even more sleek than finding my element by id or so (which I wasn't capable of)? Maybe there is even some key in the data object that tells if changes where "saved" (which also, I couldn't find)?

I don't need a precise solution. Even a hint such as "adapt some stuff in tina library yourself" or "do xx in place yy to achieve zz" would be helpful. I couldn't find any hints in the docs and I don't want to over-engineer here or do silly stuff. Thanks for the help y'all!

My state:
`const [shouldPatchDocument, setShouldPatchDocument] = useState<boolean>(
    data && placeholders ? true : false
  );`

My effect:
`useEffect(() => {
    if (shouldPatchDocument) {
       /* to dome stuff here */
      setShouldPatchDocument(false);
    }
  }, [shouldPatchDocument]);`

What I want to achieve "visualised":
<img width="430" alt="goal" src="https://github.com/tinacms/tinacms/assets/41554571/589a0bbe-4e89-40dd-aac3-639f9558ee07">
